pkg/cluster/storageaccounts.go

```diff
@@ -5,18 +5,91 @@ package cluster
 
 import (
 	"context"
+	"strings"
 
+	mgmtnetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2020-08-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/arm"
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
 )
 
+// enableServiceEndpoints should enable service endpoints on
+// subnets for storage account access
+func (m *manager) enableServiceEndpoints(ctx context.Context) error {
+	subnets := []string{
+		m.doc.OpenShiftCluster.Properties.MasterProfile.SubnetID,
+	}
+
+	for _, wp := range m.doc.OpenShiftCluster.Properties.WorkerProfiles {
+		subnets = append(subnets, wp.SubnetID)
+	}
+
+	for _, subnetId := range subnets {
+		subnet, err := m.subnet.Get(ctx, subnetId)
+		if err != nil {
+			return err
+		}
+
+		var changed bool
+		for _, endpoint := range api.SubnetsEndpoints {
+			var found bool
+			if subnet != nil && subnet.ServiceEndpoints != nil {
+				for _, se := range *subnet.ServiceEndpoints {
+					if strings.EqualFold(*se.Service, endpoint) &&
+						se.ProvisioningState == mgmtnetwork.Succeeded {
+						found = true
+					}
+				}
+			}
+			if !found {
+				if subnet.ServiceEndpoints == nil {
+					subnet.ServiceEndpoints = &[]mgmtnetwork.ServiceEndpointPropertiesFormat{}
+				}
+				*subnet.ServiceEndpoints = append(*subnet.ServiceEndpoints, mgmtnetwork.ServiceEndpointPropertiesFormat{
+					Service:   to.StringPtr(endpoint),
+					Locations: &[]string{"*"},
+				})
+				changed = true
+			}
+		}
+		if changed {
+			err := m.subnet.CreateOrUpdate(ctx, subnetId, subnet)
```

Comment:
should we `CreateOrUpdateAndWait` here? could there exist a race condition where this request has not completed,but the rp is attempting to use the service endpoint?

Reply:
how many subnets on average would this loop be going through? would we run into queuing too many requests without waiting for the previous to finish?

Reply:
Thanks for the clarification. It quells my concern here.